core/src/main/java/org/apache/iceberg/rest/responses/RESTCatalogConfigResponse.java

```diff
@@ -0,0 +1,154 @@
+  }
+
+  /**
+   * Properties that should be used to override client configuration. {@code overrides} have the highest priority
+   * and should be applied after defaults and any client-provided configuration properties.
+   *
+   * @return properties that should be given higher precedence than any client provided input
+   */
+  public Map<String, String> overrides() {
+    return overrides != null ? Maps.newHashMap(overrides) : Collections.emptyMap();
+  }
+
+  /**
+   * Merge client-provided config with server side provided configuration to return a single
+   * properties map which will be used for instantiating and configuring the REST catalog.
+   *
+   * @param clientProperties - Client provided configuration
+   * @return Merged configuration, with precedence in the order overrides, then client properties, and then defaults.
+   */
+  public Map<String, String> merge(Map<String, String> clientProperties) {
+    Preconditions.checkNotNull(clientProperties,
+        "Cannot merge client properties with server-provided properties. Invalid client configuration: null");
+    Map<String, String> merged = Maps.newHashMap(defaults);
+    merged.putAll(clientProperties);
+    merged.putAll(overrides);
+    return merged;
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("defaults", defaults)
+        .add("overrides", overrides)
+        .toString();
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+    private final Map<String, String> defaults;
+    private final Map<String, String> overrides;
+
+    private Builder() {
+      this.defaults = Maps.newHashMap();
+      this.overrides = Maps.newHashMap();
+    }
+
+    public Builder withDefault(String key, String value) {
+      Preconditions.checkNotNull(key, "Invalid config: null");
+      defaults.put(key, value);
+      return this;
+    }
+
+    public Builder withOverride(String key, String value) {
+      Preconditions.checkNotNull(key, "Invalid config: null");
+      overrides.put(key, value);
+      return this;
+    }
+
+    /**
+     * Adds the passed in map entries to the existing `defaults` of this Builder.
+     */
+    public Builder withDefaults(Map<String, String> defaultsToAdd) {
+      Preconditions.checkNotNull(defaultsToAdd, "Invalid configuration properties map: null");
+      Preconditions.checkArgument(!defaultsToAdd.containsKey(null), "Invalid config: null");
+      defaults.putAll(defaultsToAdd);
+      return this;
+    }
+
+    /**
+     * Adds the passed in map entires to the existing `overrides` of this Builder.
+     */
+    public Builder withOverrides(Map<String, String> overridesToAdd) {
+      Preconditions.checkNotNull(overridesToAdd, "Invalid configuration properties map: null");
+      Preconditions.checkArgument(!overridesToAdd.containsKey(null), "Invalid config: null");
```

Comment:
Minor: You may want to use "override" or "property" instead of just "config".

Reply:
Updated to say `overrride` and `default` for each.